### PR TITLE
[gui] Set "anywhere on path" in URL

### PR DESCRIPTION
**Pull request by bruntib**
File filter and source component filters have an "anywhere on report path" option in the web UI. The state of this option is now saved in the URL.

### CODE REVIEW

File: web/server/vue-cli/src/components/Report/ReportFilter/Filters/BaseSelectOptionFilter.mixin.js [hunk 1]
```diff
@@ -46,7 +46,7 @@ export default {
 
     getIconClass() {},
 
-    initByUrl() {
+    initOptionsByUrl() {
```
Comment (Szelethus) [inline web/server/vue-cli/src/components/Report/ReportFilter/Filters/BaseSelectOptionFilter.mixin.js:49]: More like `initCheckboxOptionsByUrl`?

**Review by Szelethus — APPROVED** [review]
LGTM! I had a single inline comment, I leave it to you whether you think that should be changed or not.

**Review by Szelethus — APPROVED** [review]
LGTM

**Review by vodorok — APPROVED** [review]
LGTM

**Pull request merged**